src/open_producten/producttypen/serializers/producttype.py

```diff
@@ -0,0 +1,160 @@
+    #     many=True,
+    #     write_only=True,
+    #     queryset=Contact.objects.all(),
+    #     default=[],
+    #     source="contacten",
+    # )
+
+    vragen = VraagSerializer(many=True, read_only=True)
+    prijzen = PrijsSerializer(many=True, read_only=True)
+    links = LinkSerializer(many=True, read_only=True)
+    bestanden = BestandSerializer(many=True, read_only=True)
+
+    class Meta:
+        model = ProductType
+        fields = "__all__"
+
+    def validate_onderwerp_ids(self, onderwerpen: list[Onderwerp]) -> list[Onderwerp]:
+        if len(onderwerpen) == 0:
+            raise serializers.ValidationError(
+                _("Er is minimaal één onderwerp vereist.")
+            )
+        return onderwerpen
+
+    def _handle_relations(
+        self,
+        *,
+        instance,
+        onderwerpen: list[Onderwerp],
+        # locaties: list[Locatie],
+        # organisaties: list[Organisatie],
+        # contacten: list[Contact],
+    ) -> None:
+        errors = dict()
+        if onderwerpen is not None:
+            build_array_duplicates_error_message(onderwerpen, "onderwerp_ids", errors)
+            instance.onderwerpen.set(onderwerpen)
+        # if locaties is not None:
+        #     build_array_duplicates_error_message(locaties, "locatie_ids", errors)
+        #     instance.locations.set(locaties)
+        # if organisaties is not None:
+        #     build_array_duplicates_error_message(
+        #         organisaties, "organisatie_ids", errors
+        #     )
+        #     instance.organisations.set(organisaties)
+        # if contacten is not None:
+        #     build_array_duplicates_error_message(contacten, "contact_ids", errors)
+        #     instance.contacten.set(contacten)
+
+        if errors:
+            raise serializers.ValidationError(errors)
+
+    @transaction.atomic()
+    def create(self, validated_data):
+        onderwerpen = validated_data.pop("onderwerpen")
+        # locaties = validated_data.pop("locaties")
+        # organisaties = validated_data.pop("organisaties")
+        # contacten = validated_data.pop("contacten")
+
+        product_type = ProductType.objects.create(**validated_data)
+
+        self._handle_relations(
+            instance=product_type,
+            onderwerpen=onderwerpen,
+            # locaties=locaties,
+            # organisaties=organisaties,
+            # contacten=contacten,
+        )
+        product_type.clean()
+        product_type.save()
+
+        return product_type
+
+    @transaction.atomic()
+    def update(self, instance, validated_data):
+        onderwerpen = validated_data.pop("onderwerpen", None)
+        # locaties = validated_data.pop("locaties", None)
+        # organisaties = validated_data.pop("organisaties", None)
+        # contacten = validated_data.pop("contacten", None)
+
+        instance = super().update(instance, validated_data)
+        self._handle_relations(
```

Comment:
Will this always overwrite the existing ones? Is it possible to just add them or is it even possible/non convoluted to program?

Reply:
Yes, it will always overwrite, in the prototype i experimented with something like onderwerp_ids_add: [] & onderwerp_ids_remove: [] inside the update requests but that seemed worse.